src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackageCommandRunner.cs

```diff
@@ -0,0 +1,362 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Evaluation;
+using NuGet.CommandLine.XPlat.Utility;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+/// <summary>
+/// A struct to simplify holding all of the information
+/// about a package reference when using list
+/// </summary>
+public struct PRPackage
+{
+    public string package;
+    public VersionRange requestedVer;
+    public string requestedVerStr;
+    public NuGetVersion resolvedVer;
+    public NuGetVersion suggestedVer;
+    public bool deprecated;
+    public bool autoRef;
+}
+
+namespace NuGet.CommandLine.XPlat
+{
+    public class ListPackageCommandRunner : IListPackageCommandRunner
+    {
+        private bool _autoReferenceFound = false;
+
+        public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs, MSBuildAPIUtility msBuild)
+        {
+
+            //If the given file is a solution, get the list of projects
+            //If not, then it's a project, which is put in a list
+            var projectsPaths = Path.GetExtension(listPackageArgs.Path).Equals(".sln")
```

Comment:
what if `path` is not defined? what kind of validations are there on input values?

Reply:
If the user does not give a path or gives a folder, the dotnet cli code will find the specific file, and send a full path, and makes sure it exists. Additionally, in ListPackageArgs, there is a check that throws an error in case path was null